src/deluge/gui/ui/keyboard_screen.cpp

```diff
@@ -383,11 +409,21 @@ int KeyboardScreen::buttonAction(int x, int y, bool on, bool inCardRoutine) {
 
 void KeyboardScreen::selectEncoderAction(int8_t offset) {
 	InstrumentClipMinder::selectEncoderAction(offset);
+	instrumentClipView.scrollVertical(0, 0, false);
+	uiNeedsRendering(this, 0xFFFFFFFF, 0);
```

Comment:
this looks like debug code?  Especially calling this without properly piping in `inCardRoutine` seems dangerous.

Reply:
Didn't notice this change, I'll check it's impact

Reply:
This just marks that the rows need rendering, the actual rendering is done later when doAnyPendingUIRendering is called